impl/src/main/java/org/ehcache/impl/internal/store/tiering/CompoundCachingTier.java

```diff
@@ -118,6 +118,26 @@ public synchronized Throwable fillInStackTrace() {
     }
   }
 
+  @Override
+  public Store.ValueHolder<V> getOrDefault(K key, Function<K, Store.ValueHolder<V>> source) throws StoreAccessException {
+    try {
+      return higher.getOrDefault(key, keyParam -> {
+        try {
+          Store.ValueHolder<V> valueHolder = lower.get(keyParam);
```

Comment:
I *think* I'm better off reporting the exception. Applying the knowledge of the calling behavior, if we return the default on seeing an exception we might end up returning illegal state from the iterator (because the inaccessible caching layer would have vetoed it).